src/cli/telemetry/telemetry.ts

```diff
@@ -0,0 +1,293 @@
+  };
+
+  await sendMetric('stencil_cli_command', data);
+  logger.debug(`${logger.blue('Telemetry')}: ${logger.gray(JSON.stringify(data))}`);
+
+  if (error) {
+    throw error;
+  }
+}
+
+function isUsingYarn() {
+  return process.env.npm_execpath.includes('yarn');
+}
+
+async function getActiveTargets(config: Config): Promise<string[][] | OutputTargetOptions[][]> {
+  return (config.outputTargets as any[]).map(target => {
+    let outputTargetSettings: OutputTargetOptions = {};
+
+    if (target.type === 'www') {
+      outputTargetSettings.serviceWorker = isObject(target?.serviceWorker);
+    }
+
+    return [target.type, outputTargetSettings];
+  });
+}
+
+async function getInstalledPackages() {
+  try {
+    // Read package.json and package-lock.json
+    const appRootDir = process.cwd();
+
+    const packageJson: PackageJsonData = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package.json'),
+    );
+    const packageLockJson: any = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package-lock.json'),
+    );
+
+    const packages = Object.entries({
+      ...packageJson.devDependencies,
+      ...packageJson.dependencies,
+    });
+
+    // Collect packages only in the stencil, ionic, or capacitor org's:
+    // https://www.npmjs.com/org/stencil
+    const ionicPackages = packages.filter(
+      ([k]) => k.startsWith('@stencil/') || k.startsWith('@ionic/') || k.startsWith('@capacitor/'),
+    );
+
+    // Original method, but cross reference with package-lock.json, don't consider yarn.
+    const versions = packageLockJson
+      ? ionicPackages.map(
+          ([k, v]) =>
+            `${k}@${packageLockJson.dependencies[k].version || packageLockJson.devDependencies[k].version || v}`,
+        )
+      : ionicPackages.map(([k, v]) => `${k}@${v}`);
+
+    return versions;
+  } catch (err) {
+    hasDebug() && console.error(err);
+    return [];
+  }
+}
+
+export async function shouldTrack(ci: boolean) {
+  return (await checkTelemetry()) && !ci && isInteractive();
+}
+
+export async function getTelemetryToken() {
+  const config = await readConfig();
+  return config['tokens.telemetry'];
+}
+
+export async function checkTelemetry() {
+  const config = await readConfig();
+  return config['telemetry.stencil'];
+}
+
+export async function enableTelemetry() {
+  await updateConfig({ 'telemetry.stencil': true });
+}
+
+export async function disableTelemetry() {
+  await updateConfig({ 'telemetry.stencil': false });
+}
+
+/**
+ * If telemetry is enabled, send a metric via IPC to a forked process for uploading.
+ */
+export async function sendMetric<D>(name: string, data: D): Promise<void> {
+  const message: Metric<string, D> = {
+    name,
+    timestamp: new Date().toISOString(),
+    source: 'stencil_cli',
+    value: data,
+  };
+
+  await sendTelemetry({ type: 'telemetry', message });
+}
+
+export async function sendTelemetry(data: any) {
```

Comment:
This function only appears to be used in this file. I think we can safely remove `export`:
```suggestion
async function sendTelemetry(data: any) {
```